src/main/java/core/basesyntax/operation/ReturnHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class ReturnHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.RETURN) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopServiceImpl.addFruits(fruit, quantity);

```

Comment:
It seems like you're not checking if the quantity of the fruit is positive before adding it to the shop. According to the task checklist, you should always check the balance before saving it in the Storage. It should be positive. Throw a `RuntimeException` if the balance is negative.